### ISSUE TITLE
Add citations to the webapp

### ISSUE THREAD

**Issue by alsiesta** [Azure-Samples/chat-with-your-data-solution-accelerator#1106]
# Motivation

There is other Azure Chats where the WebApp Frontend offers the feature that links to the matching documents are offered in the completion, you can click them and in a side panel the related embedding shows up and a link to the data file.

**Comment by adamdougal** [comment]
@alsiesta Heya! Can you elaborate a bit further? Citations are already present on the deployed web app frontend.

**Comment by alsiesta** [comment]
Yes I finally realized that citations can be activated, if I set the CONVERSATION_FLOW=byod
But that is not abvious. The markdown doc. https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/blob/main/docs%2Fconversation_flow_options.md doesn't even mention, that this will enable the citations.

I wonder, what other great features are still hidden 😀

It would for instance be great to the chat history.

Kind regards and thank you, for feedback,  Alexander
________________________________

**Comment by adamdougal** [comment]
Citations should be returned regardless of  what `CONVERSATION_FLOW` is set to. This is what I see for example when it's set to `custom` which is the default:

![image](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/assets/1589887/4575608f-57f3-431f-8709-bd74717e425a)

Do you not see this at all when it's set to `custom`?

**Comment by alsiesta** [comment]
Short answer: I don't see any citations when CONVERSATION_FLOW is set to 'custom'.
I found out by chance, that when CONVERSATION_FLOW is set to 'byod' only I do see them - that's why i posted my comment, because its a super important feature. I would rather set 'byod' as default than 'custom'. Because asintermediate programmer like me, who has done a lot, there is still much more i haven't done or know about yet. 
I never heard about a custom orchestrator - i do see it is complex - but to start with, I first have to learn about it.

**Comment by amer7862000** [comment]
I'm not sure if this is directly related, but could we include a reference to the specific PDF page where the information was sourced? Right now, the PDF link opens the document but it doesn't navigate to the exact page. Hope that makes sense.
